### PR TITLE
Fixed zoom out mode in edit template

### PR DESCRIPTION
**Pull request by PARTHVATALIYA**
## What?
Part of - https://github.com/WordPress/gutenberg/issues/65505

## Why?
Solve - https://github.com/WordPress/gutenberg/issues/65505

## Testing Instructions
1. Open the site editor
2. Go to Pages > any page to edit a page within the site editor
3. Click the Zoom out button in the top bar
4. Double click the post title block
5. Notice that you will edit the post title by double clicking on it.

## Screenshots or screencast 
[Pages ‹ Template ‹ gutenberg ‹ Editor — WordPress.webm](https://github.com/user-attachments/assets/b5c492e9-b97f-4eca-a5e3-e7d203725657)

### CODE REVIEW

**Comment by colorful-tones** [comment]
Thanks for the contribution @PARTHVATALIYA !

I tested this out and the fix works. I can not speak to the code, but it addresses the issue.

File: packages/editor/src/components/visual-editor/edit-template-blocks-notification.js [hunk 1]
```diff
@@ -27,16 +28,21 @@ import { store as editorStore } from '../../store';
...
-			onNavigateToEntityRecord:
-				getEditorSettings().onNavigateToEntityRecord,
-			templateId: getCurrentTemplateId(),
-		};
-	}, [] );
+			return {
+				onNavigateToEntityRecord:
+					getEditorSettings().onNavigateToEntityRecord,
+				templateId: getCurrentTemplateId(),
+				editorMode: 'zoom-out' === __unstableGetEditorMode(),
```
Comment (draganescu) [inline packages/editor/src/components/visual-editor/edit-template-blocks-notification.js]: This looks like it's bettter named `isZoomOutMode`

**Review by draganescu — COMMENTED** [review]
Thanks for working on this. I find it strange that the notification component knows all this. Can we find a way to never have the notification if we're in zoom out mode?

File: packages/editor/src/components/visual-editor/edit-template-blocks-notification.js (hunk 1 above)
Comment (PARTHVATALIYA) [reply packages/editor/src/components/visual-editor/edit-template-blocks-notification.js]: Changed to `isZoomOutMode`

**Comment by PARTHVATALIYA** [comment]
> Thanks for working on this. I find it strange that the notification component knows all this. Can we find a way to never have the notification if we're in zoom out mode?

Thanks @draganescu for reviewing the PR. I am new to gutenberg can you please explain me more. So I can resolve this.

**Comment by draganescu** [comment]
The EditTemplateBlocksNotification component should only show a notification, not also set editor modes and this PR adds too much to it.

What I mean is that instead of editing the EditTemplateBlocksNotification component to not render if zoom out and to also set the editor mode, we need to find where the  EditTemplateBlocksNotification component is rendered and render it there conditionally depending on editor mode. Also I suspect we won't have to reintercept double click since thre is something else already setting editor mode on double click, the bug to solve is to not show this notification. 

Not sure if I conveyed it right, is it any clearer? 😀

**Comment by PARTHVATALIYA** [comment]
Thank you for explanation! Issue resolved.

**Comment by andrewserong** [comment]
Thank you for working on this @PARTHVATALIYA! Apologies, I missed the ping earlier on this change and I submitted a very similar PR over in #65963. Are you keen to continue working on this one? If not, I'm happy to either jump in here, or continue on with the similar PR over in #65963.

It seems the main next step is to look into the feedback from @draganescu regarding where we place this check, as arguably it'd be better if we didn't include the logic within the `EditTemplateBlocksNotification` component itself. Perhaps we could look into skipping using that notification component if zoom out mode is engaged, and add a double click handler somewhere else for checking whether to switch off zoom out mode (as suggested in https://github.com/WordPress/gutenberg/pull/65565#issuecomment-2396659226).

What do you think?

**Comment by getdave** [comment]
Could we close this one out and then include @PARTHVATALIYA in props on your PR @andrewserong?

**Comment by PARTHVATALIYA** [comment]
@andrewserong you can continue with the PR https://github.com/WordPress/gutenberg/pull/65963.

**Comment by andrewserong** [comment]
Thanks for confirming @PARTHVATALIYA, and thanks for your work here, I'll be sure to include props for you over in #65963 when it lands!

**Comment by getdave** [comment]
Closing in favour of https://github.com/WordPress/gutenberg/pull/65963.

**Pull request closed without merging**